de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/service/IBPartnerStatsBL.java

```diff
@@ -34,16 +39,36 @@
  */
 public interface IBPartnerStatsBL extends ISingletonService
 {
+	/**
+	 */
+	@Builder
+	@Value
+	public static class CalculateSOCreditStatusRequest
+	{
+		@NonNull
+		BPartnerStats stat;
+		@NonNull
+		@Default
+		BigDecimal additionalAmt = BigDecimal.ZERO;
+		@NonNull
+		Timestamp date;
+		@Default
+		boolean forceCheckCreditStatus = false;
+	}
+
 	/**
 	 * Calculate the future/simulated SOCreditStatus for the given {@link BPartnerStats} object at a certain date
-	 * No updating
+	 * <br>
+	 * The computation can be forced with the flag <code>forceCheckCreditStatus</code><br>
+	 * If the status is <code>CreditStop</code>, the status can be recomputed only if flag <code>forceCheckCreditStatus</code> is on Y
+	 * <br><b>No updating</b>
 	 *
 	 * @param stat
 	 * @param additionalAmt
 	 * @param date
 	 * @return
 	 */
-	String calculateSOCreditStatus(BPartnerStats stat, BigDecimal additionalAmt, Timestamp date);
+	String calculateSOCreditStatus(CalculateSOCreditStatusRequest request);
```

Comment:
I would name the method `calculateProjectedSOCreditStatus` because actually that's what it does.